src/appleseed.studio/mainwindow/rendering/renderwidget.cpp

```diff
@@ -422,5 +430,31 @@ void RenderWidget::paintEvent(QPaintEvent* event)
     m_painter.end();
 }
 
+void RenderWidget::dragEnterEvent(QDragEnterEvent* event)
+{
+    if (event->mimeData()->hasFormat("text/plain"))
+        event->acceptProposedAction();
+}
+
+void RenderWidget::dragMoveEvent(QDragMoveEvent* event)
+{
+    if (pos().x() <= event->pos().x() && pos().y() <= event->pos().y()
```

Comment:
Formatting proposal for improved readability:

    if (pos().x() <= event->pos().x() &&
        pos().y() <= event->pos().y() &&
        event->pos().x() < pos().x() + width() &&
        event->pos().y() < pos().y() + height())

Reply:
this part is not finish, if I leave the scene by the left side, the icon change to show that drop is not available but if you continue it come back to the drop-available icon.
I need to understand why and fix it before closing this request

Reply:
to give a little idea, inside the scene x change between 0 and 700, and when I live from the left (so x was 0), x become 700 again and decrement when you go futher to the left
I dont really see a way to fix this...

Reply:
That's weird. There must be a way. Did you check the docs carefully? What about posting a question on Stack Overflow or on Qt Centre?

Reply:
Also, did you check carefully how `QDragMoveEvent` works? https://doc.qt.io/archives/qt-4.8/qdragmoveevent.html#details

Reply:
I tried accept(const QRect & rectangle) but it is worth
I dont understand why the drag and drop behave like this